src/Articulate/Routing/ArticulateRouter.cs

```diff
@@ -70,9 +70,12 @@ public void MapRoutes(HttpContext httpContext, IUmbracoContext umbracoContext)
 
             var domains = umbracoContext.Domains.GetAll(false).ToList();
 
-            // TODO: Enable this in some way
-            // clear the existing articulate routes (if any)
-            // RemoveExisting(routes);
+            // Ensure we always start with an empty cache
+            // We may call this MapRoutes method again when Articulate root node is published
+            // and any of the dynamic URLs from the content node change
+            // So we clear this out, otherwise we will have the previous working URL and the updated URL (Until the site restarts)
+            _routeCache.Clear();
```

Comment:
I think all of MapRoutes will need to be put inside a c# lock since it's not going to be thread safe

Reply:
Noob question @Shazwazza how can you tell this will not be thread safe and that I need to add a lock around all the code in this method ?

Reply:
Well if 2 or more threads were to call this API at the same time what do you think will happen? We are manipulating multiple things here, so one thread could get as far as populating the routes while the other thread clears them and then weird things happen. There's all sorts of problems that could occur, like if 2 or more threads get to the point of writing the routes, we'll probably get duplicate dictionary key errors, etc...